src/pages/index.js

```diff
@@ -38,54 +89,113 @@ const createCard = (cardData) => {
 const addCardObject = {
   popupSelector: "#add-card-modal",
   handleFormSubmit: (inputValues) => {
-    const cardElement = createCard(inputValues);
-    addCardForm.popupForm.reset();
-    addCardFormValidator.toggleButtonState();
-    cardSelection.addItem(cardElement);
-    addCardForm.close();
+    api
+      .addCard(inputValues.name, inputValues.link)
+      .then((res) => {
+        // cardElement = createCard(inputValues);
+        // const cardElement = createCard(inputValues);
+        const cardElement = createCard(res);
+        // addCardForm.popupForm.reset();
+        // addCardFormValidator.toggleButtonState();
+        cardSelection.addItem(cardElement);
+        addCardForm.close();
+      })
+      .finally(() => {
+        addCardForm.popupForm.reset();
+        addCardFormValidator.toggleButtonState();
```

Comment:
You need to **reset** the form only  in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more


**Please, correct it everywhere**

Reply:
I removed all the finally blocks